thrift/fields.go

```diff
@@ -29,35 +29,53 @@ import (
 	"github.com/thriftrw/thriftrw-go/wire"
 )
 
-func fieldMap(fields compile.FieldGroup) map[string]*compile.FieldSpec {
-	m := make(map[string]*compile.FieldSpec)
+// fieldMap returns maps from string to the field spec.
+// The first map is an exact map, which uses the name as specified in the Thrift file.
+// The second map is a fuzzy map, which will ignore case, and ignore
+// any non-alphanumberic characters.
+func fieldMap(fields compile.FieldGroup) (exact, fuzzy map[string]*compile.FieldSpec) {
+	exact = make(map[string]*compile.FieldSpec)
+	fuzzy = make(map[string]*compile.FieldSpec)
 	for _, f := range fields {
-		m[f.ThriftName()] = f
+		exact[f.ThriftName()] = f
+		fuzzy[fuzz(f.ThriftName())] = f
 	}
-	return m
+
+	// If there's any field clashes in the fuzzy map, skip fuzzy matching.
+	if len(exact) != len(fuzzy) {
+		fuzzy = nil
+	}
+
+	return exact, fuzzy
 }
 
 func fieldGroupToValue(fieldsList compile.FieldGroup, request map[string]interface{}) ([]wire.Field, error) {
 	var (
-		fields = fieldMap(fieldsList)
+		exactFields, fuzzyFields = fieldMap(fieldsList)
 
-		err = fieldGroupError{available: sorted.MapKeys(fields)}
+		err = fieldGroupError{available: sorted.MapKeys(exactFields)}
 
 		// userFields is the user-specified values by field name.
 		userFields = make(map[string]interface{})
 	)
 
 	for k, v := range request {
-		_, ok := fields[k]
+		_, ok := exactFields[k]
 		if !ok {
-			err.addNotFound(k)
-			continue
+			// Try a fuzzy match
+			field, ok := fuzzyFields[fuzz(k)]
```

Comment:
`nil` maps are actually valid empty maps (for reading). Since these are immutable maps, it works fine:
http://play.golang.org/p/0L0CzDAk_Y